### DISCUSSION TITLE
Enable all possible user agents by default

### DISCUSSION THREAD

**Discussion by shahidjalally** [tarampampam/random-user-agent#463] [General] [answered]
Hello folk,

I am using this chrome extension and found it is good enough for my project.
However I am looking a solution if I could become able to enable all user agents  by default in options.js file before unpack / or install this extension.

actually I am using it in python and selenium script and I need enabled all user agents by default.

If some one could help me in this regard, I will be very thankful to him/her.

Finally I am looking forward to hearing from you positively.

Regards.

**Comment by tarampampam** [comment]
Hi there! You can build the extension with all required default options, this is very simple (watch the Makefile source)

**Reply by shahidjalally** [reply]
hello tarampampam,
I am new in this and still confused and stuck here.
please help me more or give me more easy clue.

**Reply by tarampampam** [reply, marked as answer]
- Clone the repository
- Edit here https://github.com/tarampampam/random-user-agent/blob/0af4a288c2a8734652f04c32d57959fe7a680119/src/settings/settings.ts#L97
- Remove this https://github.com/tarampampam/random-user-agent/blob/0af4a288c2a8734652f04c32d57959fe7a680119/src/settings/settings.ts#L114-L126
- run `yarn build`

**Reply by shahidjalally** [reply]
@tarampampam thank you budy.
I tried it, some I installed yarn dependencies (npm, yaran & webpack module etc).
And finally got success to build this extension after these changes.

Now I need an other favour. still safari browser and andriod OS generatortype.xx codes are missing.
If you could provide me or guide me how may I Include safari borwser and andriod OS into #line 97 array it will be great help.

**Reply by tarampampam** [reply]
You can find all supported types here:

https://github.com/tarampampam/random-user-agent/blob/0af4a288c2a8734652f04c32d57959fe7a680119/src/useragent/generator.ts#L15-L28

**Reply by shahidjalally** [reply]
@tarampampam thank you man!
really good support.
Thumbs Up

**Reply by tarampampam** [reply]
You are welcome!

**Reply by jackma81** [reply]
I have the same question as him, but now the source code has been updated. Can you help me complete it? Thank you very much

**Reply by tarampampam** [reply]
> I have the same question as him, but now the source code has been updated. Can you help me complete it? Thank you very much

It's as simple as before, edit here:

https://github.com/tarampampam/random-user-agent/blob/59647fa13620ebbc37ed8118c501a8ed947f1fab/src/entrypoints/background/persistent/settings.ts#L15-L43

Remove this one:

https://github.com/tarampampam/random-user-agent/blob/59647fa13620ebbc37ed8118c501a8ed947f1fab/src/entrypoints/background/persistent/settings.ts#L52-L64

And run `npm run build`

**Reply by jackma81** [reply]
Thank you man!
Really good support.
With love 🫶
